tests/unit/backend/corpora/common/entities/test_dataset.py

```diff
@@ -47,6 +47,34 @@ def test__get__does_not_exist(self):
         non_existent_key = "non_existent_id"
         self.assertEqual(Dataset.get(non_existent_key), None)
 
+    def create_dataset_with_artifacts(self, artifact_count=1, deployment_dir_count=1, artifact_params=None):
+        """
+        Create a dataset with a variable number of artifacts, and deployment_directories
+        """
+        if not artifact_params:
+            artifact_params = dict(
+                filename="filename_x",
+                filetype=DatasetArtifactFileType.H5AD,
+                type=DatasetArtifactType.ORIGINAL,
+                user_submitted=True,
+                s3_uri="some_uri",
+            )
+
+        deployment_directory_params = dict(url="test_url")
+
+        dataset_params = BogusDatasetParams.get()
+        dataset = Dataset.create(
```

Comment:
use generate_dataset to make sure everything gets cleaned up after the test or make use of TestCase.addcleanup()